pyspecdata/core.py

```diff
@@ -1617,21 +1618,34 @@ def concat(datalist, dimname, chop=False):
         )
     other_info_out = datalist[0].other_info
     if dimname in datalist[0].dimlabels:
+        dim_idx = datalist[0].axn(dimname)
         assert all(
-            [dimname in datalist[j].dimlabels for j in range(len(datalist))]
-        ), (
-            "the dimension '"
-            + dimname
-            + "' exists in the first datset, but not the second!"
-        )
+            [
+                datalist[j].dimlabels == datalist[0].dimlabels
+                for j in range(len(datalist))
+            ]
+        ), "the dimlabels for all your datasets do no match and/or are not ordered the same way"
+        # {{{ check that all the shapes match, too, aside from the dim we concat along
+        shape_check = [
+            list(datalist[j].data.shape) for j in range(len(datalist))
+        ]
+        for j in shape_check:
+            j.pop(dim_idx)
+        assert all(
+            [shape_check[j] == shape_check[0] for j in range(len(shape_check))]
+        ), ("shapes " + str(shape_check) + " are not all equal")
+        # }}}
```

Comment:
reading it again, I realized that we probably wanted more error checks to avoid weird behavior.  Please read these and make sure you're cool with them, and that the example runs the way you wnat